liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,404 @@
+    data.totalNewAccounts += data.newAccounts.length;
+    data.newAccounts = [];
+    // #endregion
+
+    // #region Update Balances on zero health accounts
+    const filteredAccounts = [];
+    Object.keys(accounts).forEach((currentAccount) => {
+      if (accounts[currentAccount].health == null
+        || accounts[currentAccount].health === 0) {
+        filteredAccounts.push(currentAccount);
+        // Zero account balances
+        Object.keys(supply).forEach((currentToken) => {
+          supply[currentToken][currentAccount] = null;
+        });
+        Object.keys(borrow).forEach((currentToken) => {
+          borrow[currentToken][currentAccount] = null;
+        });
+      }
+    });
+    /* eslint-enable no-param-reassign */
+
+    // Grab the assets in first, and make sure they are initialized
+    const foundTokens = [];
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      const assetsIn = await comptrollerContract.getAssetsIn(currentAccount);
+      accounts[currentAccount].assetsIn = assetsIn.map((asset) => {
+        const address = asset.toLowerCase();
+        foundTokens.push(address);
+        return address;
+      });
+    }));
+    // Use Array to Set to Array to remove duplicates
+    const filteredTokens = Array.from(new Set(foundTokens));
+
+    // Initialize token objects
+    await Promise.all(filteredTokens.map(async (token) => {
+      await verifyToken(data, token);
+    }));
+
+    // Grab token balances from on-chain
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      await Promise.all(accounts[currentAccount].assetsIn.map(async (currentToken) => {
+        const snapshot = await tokens[currentToken].contract.getAccountSnapshot(currentAccount);
+
+        // update the supply balance and token quantity
+        supply[currentToken][currentAccount] = BigNumber(snapshot[1].toString())
+          .dividedBy(tokens[currentToken].cTokenDecimalsMult);
+
+        // update the borrow balance
+        borrow[currentToken][currentAccount] = BigNumber(snapshot[2].toString());
+      }));
+    }));
+    // #endregion
+
+    // #region Update all token prices via 1inch for now.
+    // Mapping through all tokens
+    // Note: Object.entries does not work here because the nested objects cannot be retrieved.
+    await Promise.all(Object.keys(tokens).map(async (currentToken) => {
+      const price = await oneInchContract.getRateToEth(tokens[currentToken].underlying, 0);
+      const oneInchMult = BigNumber(10).pow(36 - tokens[currentToken].tokenDecimals);
+
+      // Adjust for native decimals
+      tokens[currentToken].price = BigNumber(price.toString()).dividedBy(oneInchMult);
+
+      // Update the Collateral Factor
+      const market = await comptrollerContract.markets(currentToken);
+      tokens[currentToken].collateralMult = BigNumber(market[1].toString())
+        .dividedBy(BigNumber(10).pow(18));
```

Comment:
Add a comment explaining that the collateralFactorMantissa (`market[1]`) is scaled by `1e18`